prow/crier/reporters/github/reporter.go

```diff
@@ -162,8 +165,61 @@ func (c *Client) Report(ctx context.Context, log *logrus.Entry, pj *v1.ProwJob)
 		defer lock.Release(1)
 	}
 
+	pjsToReport := func() []v1.ProwJob {
+		if len(pj.Spec.Refs.Pulls) != 1 {
+			return nil
+		}
+		// find all prowjobs from this PR
+		selector := map[string]string{}
+		for _, l := range []string{kube.OrgLabel, kube.RepoLabel, kube.PullLabel} {
+			selector[l] = pj.ObjectMeta.Labels[l]
+		}
+		var pjs v1.ProwJobList
+		if err := c.lister.List(ctx, &pjs, ctrlruntimeclient.MatchingLabels(selector)); err != nil {
+			log.WithError(err).Errorf("Cannot list prowjob with selector %v", selector)
+			return nil
+		}
+
+		latestBatch := make(map[string]v1.ProwJob)
+		for _, pjob := range pjs.Items {
+			if len(pjob.Spec.Refs.Pulls) != 1 {
+				continue
+			}
+			if pj.Spec.Refs.Pulls[0].SHA != pjob.Spec.Refs.Pulls[0].SHA {
+				continue
+			}
+			// Now you have convinced me that you are the same job from my revision,
+			// continue convince me that you are the last one of your kind
+			if existing, ok := latestBatch[pjob.Spec.Job]; !ok {
+				latestBatch[pjob.Spec.Job] = pjob
+			} else if pjob.CreationTimestamp.After(existing.CreationTimestamp.Time) {
+				latestBatch[pjob.Spec.Job] = pjob
+			}
+		}
+
+		// If current pj is not the latest, we should skip
+		if existing, ok := latestBatch[pj.Spec.Job]; !ok {
```

Comment:
We need the latest job for the pull request, not the latest job for a given job or not?

Reply:
this is checking for an edge condition that might only exists in my head :)

say that job `foo` with number N is about to finish, and the developer saw error log in prow UI ahead of time, and know it was a flake, commented `/test foo` and triggered a new prowjob numbered M, which was not fast enough to cancel N before it's finished. In this case crier saw both M and N, and we should not report. And actually while I'm writing this, noticed that this case is actually handled by the check later, so I think this can be deleted

Reply:
Ah ok, yeah generally this can always race with job creation.

What I meant is that we should only report the oldest job to prevent the reporter from doing the comment reporting for every job once all jobs finished.

Reply:
the idea here is to report when all running jobs are finished, did you mean latest instead of oldest?

Reply:
yeah, but what I mean is that if e.G. we have seven ProwJobs, we will end up calling `report.ReportComment` seven times once all of them finished or did I miss something? The idea was to abort if we do not have the oldest pj here to only do it once.

Reply:
The first six finished jobs will be labeled as reported by github reporter, and won't be reconciled any more unless I missed something? So in this case `report.ReportComment` will only be called on the seventh job completion. It would only be called seven times if all seven jobs completed all at the same time, which should be rare?